packages/truffle-core/lib/commands/config.js

```diff
@@ -1,8 +1,113 @@
+const parse = function(args) {
+  if (args.length === 0) {
+    return null;
+  }
+
+  let command = args[0];
+
+  if (typeof command !== "string") {
+    // invalid command
+    throw new Error(`Invalid config option "${command}"`);
+  }
+  command = command.toLowerCase();
+
+  let set = false;
+  let key = args[1];
+  let value = args[2];
+
+  switch (command) {
+    case "enable": {
+      set = true;
+      value = true;
+      break;
+    }
+    case "disable": {
+      set = true;
+      value = false;
+      break;
+    }
+    case "get":
+    case "read": {
+      set = false;
+      if (typeof key === "undefined" || key === null || key === "") {
+        // invalid key
+        throw new Error("Must provide a <key>");
+      }
+
+      break;
+    }
+    case "set":
+    case "write": {
+      set = true;
+      if (typeof key === "undefined" || key === null || key === "") {
+        // invalid key
+        throw new Error("Must provide a <key>");
+      }
+
+      if (typeof value !== "string" || value === "") {
+        // invalid value
+        throw new Error("Must provide a <value-for-set>");
+      }
+
+      switch (value.toLowerCase()) {
+        case "null": {
+          value = null;
+          break;
+        }
+        case "undefined": {
+          value = undefined;
+          break;
+        }
+        case "true": {
+          value = true;
+          break;
+        }
+        case "false": {
+          value = false;
+          break;
+        }
+        default: {
+          // check if number, otherwise leave as string
+          const float = parseFloat(value);
+          if (!isNaN(float) && value === float.toString()) {
+            value = float;
+          }
+          break;
+        }
+      }
+
+      break;
+    }
+    default: {
+      if (
+        command !== "--enable-analytics" &&
+        command !== "--disable-analytics" &&
+        command !== ""
+      ) {
+        // TODO: Deprecate the --(en|dis)able-analytics flag in favor for `enable analytics`
+        // invalid command!
+        throw new Error(`Invalid config option "${command}"`);
+      } else {
+        // we should not have gotten here
+        return null;
+      }
+    }
+  }
+
+  return {
+    set,
+    userLevel: module.exports.userLevelSettings.includes(key),
```

Comment:
Why not just `command` rather than `module.exports`? In the future I wonder if this `parse` function should be pulled out into its own utility, and then this would have to be passed in somehow. I'm in favor of these specific command files being kept as lean as possible and this file is just starting to get a bit large.

Reply:
Ah, `parse` was in the `command` so I had `module.exports` to reference itself. I can change this back to `command` now that I moved `parse` into its own function

I agree the file was getting a bit large. There wasn't a pattern yet for a multi-file `commands/*.js` command yet, so I kind of was like "ehhh, they can figure out when/if they want to change the architecture".

Reply:
Hehe yeah I think the next time this happens I'm going to just give it a folder and include helper files in there. I just wanted to note this here, I'm not going to fight now to create that architecture :) Thanks Mike!

Reply:
Fixed using `command` instead of `module.exports` in https://github.com/trufflesuite/truffle/pull/2302/commits/96a54f2e50ec8e0eafb38dfb1737c6c3562bf314